src/iface-modbus.c

```diff
@@ -170,16 +170,6 @@ static int read_u64(modbus_t *ctx, uint16_t addr, uint64_t *out)
 	return modbus_read_registers(ctx, addr, 4, (uint16_t *) out);
 }
 
-static int parse_url(const char *url)
-{
-	if (strncmp(url, TCP_PREFIX, TCP_PREFIX_SIZE) == 0)
-		return TCP;
-	else if (strncmp(url, RTU_PREFIX, RTU_PREFIX_SIZE) == 0)
-		return RTU;
-	else
-		return -EINVAL;
-}
-
```

Comment:
I would change the function behavior instead of changing its placement. I think an alternative could be this function act as an interface to create a connection, so instead of returning the connection type, it could return the context.

Reply:
done